src/caselawclient/search_result.py

```diff
@@ -0,0 +1,264 @@
+        uri = node.xpath("@uri")[0].lstrip("/").split(".xml")[0]
+        matches = SearchMatches(etree.tostring(node, encoding="UTF-8").decode("UTF-8"))
+        matches_html = matches.transform_to_html()
+        namespaces = {
+            "search": "http://marklogic.com/appservices/search",
+            "uk": "https://caselaw.nationalarchives.gov.uk/akn",
+            "akn": "http://docs.oasis-open.org/legaldocml/ns/akn/3.0",
+        }
+        neutral_citation = (
+            node.xpath("search:extracted/uk:cite", namespaces=namespaces)[0].text
+            if node.xpath("search:extracted/uk:cite", namespaces=namespaces)
+            else ""
+        )
+        court_code = (
+            node.xpath("search:extracted/uk:court", namespaces=namespaces)[0].text
+            if node.xpath("search:extracted/uk:court", namespaces=namespaces)
+            else ""
+        )
+        metadata_name = (
+            node.xpath("search:extracted/akn:FRBRname/@value", namespaces=namespaces)[0]
+            if node.xpath("search:extracted/akn:FRBRname/@value", namespaces=namespaces)
+            else ""
+        )
+        date_string = (
+            node.xpath(
+                "search:extracted/akn:FRBRdate[(@name='judgment' or @name='decision')]/@date",
+                namespaces=namespaces,
+            )[0]
+            if node.xpath(
+                "search:extracted/akn:FRBRdate[(@name='judgment' or @name='decision')]/@date",
+                namespaces=namespaces,
+            )
+            else ""
+        )
+        transformation_date = (
+            node.xpath(
+                "search:extracted/akn:FRBRdate[@name='transform']/@date",
+                namespaces=namespaces,
+            )[0]
+            if node.xpath(
+                "search:extracted/akn:FRBRdate[@name='transform']/@date",
+                namespaces=namespaces,
+            )
+            else ""
+        )
+        content_hash = (
+            node.xpath("search:extracted/uk:hash", namespaces=namespaces)[0].text
+            if node.xpath("search:extracted/uk:hash", namespaces=namespaces)
+            else ""
+        )
+
+        try:
+            date = dateparser.parse(date_string)
+        except ParserError as e:
+            logging.warning(
+                f'Unable to parse document date "{date_string}". Full error: {e}'
+            )
+            date = None
+
+        try:
+            court = courts.get_by_code(court_code)
+        except CourtNotFoundException:
+            court = None
+
+        try:
+            return SearchResult(
+                uri=uri,
+                neutral_citation=neutral_citation,
+                name=metadata_name,
+                matches=matches_html,
+                court=court,
+                date=date,
+                content_hash=content_hash,
+                transformation_date=transformation_date,
+                meta=meta,
+            )
+        except caselawclient.errors.MarklogicAPIError as e:
```

Comment:
EditorUI had the try except whereas PUI didnt.

I imagine we saw a bug in EUI and thats why we added that.


Will add a test unless people think it is unncessary to have this try except

Reply:
Remove the try except -it looked like:
```
        try:
            return SearchResult(
                uri=uri,
                neutral_citation=neutral_citation,
                name=metadata_name,
                matches=matches_html,
                court=court,
                date=date,
                content_hash=content_hash,
                transformation_date=transformation_date,
                meta=meta,
            )
        except caselawclient.errors.MarklogicAPIError as e:
            logging.warning(
                f"Unable to create search result for {uri}. Has it been deleted? Full error: {e}"
            )
            return None
```

but since I have refactored the metatadata initialisation out of this class, it is no longer needed here. Considered puttin in SearchResultsMeata.create_from_node around the apiclient calls but then I thought that the try except was definitely particular to the implementation in EUI and that the client library shouldnt decide to return a None object when SearchResultsMeta is called. If this does error, we can continue handling in eui (and maybe in PUI too).